### ISSUE TITLE
Support For Proxy Servers(Velocity,Bungeecord,etc) Please 

### ISSUE THREAD

**Issue by R3TR0V1P3R** [Skidamek/AutoModpack#224]
### Motivation

i just want to be able to connect to my proxy servers witch then connect to the server that has automodpack and for it to work, right now automodpack just says the client doesnt have it installed

### Description

so im just really looking for maybe a proxy plugin that then helps relay that kind of information correctly would be awsome or if you can find away for it to do like a double check or somthing and maybe that will fix it idk i just really want to use this for a multi server idea i have for my friends and i.

Sincerely, A diehard minecrafter:)

**Comment by Skidamek** [comment]
I am planning to make compatibility velocity plugin

**Comment by R3TR0V1P3R** [comment]
Perfect thank you so much is there any way I can make this more of a priority sorry really like your mod :)

**Comment by Joshy260** [comment]
Please do this asap 🙏

**Comment by WataLaker37821** [comment]
hi, i watched all new mod versions, can you please tell if velocity error is fixed or not yet?

**Comment by Skidamek** [comment]
> hi, i've watched all new mod versions, can you please tell if velocity error is fixed or not yet?

It's not, it's a hobby project, I don't have that much time to work on random stuff I don't even would use. This will be eventually done but it will take time. If you want it quicker you can start implementing it yourself, this is open source.

What you would actually need to do is to rewrite the server side part of this mod to velocity. This mod is very modular already (one build command builds 16 different versions for fabric/neo/forge) so this shouldn't be that hard. Code which you would mostly need is in core module which are minecraft and loader independent just standard java. However you would need to reasech how to send and receive login stage packets and how to register commands.

**Comment by WataLaker37821** [comment]
okay, im not sure if i can make it, but will take your help with glad! Thanks!

**Comment by thecodemonster-AI** [comment]
The Problem is with the way automod saves each mudpack to a server. because I have the same thing where it thinks its a different server due to the shifting ip's. A solution would be to pull the server address from the server.dat file and use that because the server address doesn't change.

**Comment by Skidamek** [comment]
> The Problem is with the way automod saves each modpack to a server. because I have the same thing where it thinks its a different server due to the shifting ip's. A solution would be to pull the server address from the server.dat file and use that because the server address doesn't change. This would work only if one server is under a proxy. For the server that have multiple modpacks under a proxy, you could have a check the mods in the server then the client and compare. if they match then send a hashed string to link the server and client, allowing the server/client to link instantly without another check.

No, its completely something else. If you want modpack to have constant name which represents it properly, just name it in a server config, it wont shift anymore even if server changes ip.

@thecodemonster-AI  How do you even got it working under a proxy in a first place? What proxy do you use? 

AutoModpack does not work under a proxies (at least not on velocity) because of login packets implementation in velocity, thus it will require separate automodpack velocity plugin (it is in works) which should be a replacement of a server side mod version from fabric/neo/forge servers.

**Comment by thecodemonster-AI** [comment]
I have been using ngrok witch works similar to other but the first start it downloads the mods then restarts. second start it want me to down load them again. I see that it is creating multiple modpack files in auto modpack. if you decline the download the mods are already loaded into the clients so it just allows you into the server. Idk how the mod detects whether the it is a different server but it will just create multiple modpack files thinking its a different server. but dealing with proxys that have multiple server running on them requires checks to make sure the server that your trying to connect to is the correct server and if it is then not to repromp downloading mods

**Comment by Skidamek** [comment]
> If you want modpack to have constant name which represents it properly, just name it in a server config `modpackName`, it wont shift anymore even if server changes ip.

> but dealing with proxys that have multiple server running on them requires checks to make sure the server that your trying to connect to is the correct server and if it is then not to repromp downloading mods

That wont be a problem since plan for plugin implementation is that it will completely replace the mod version and in the future may have additional proxy features like per server modpack but its all gonna be managed on proxy server. Mod servers fabric/neo/forge will not even know about automodpack existence in the process.

**Comment by wdp-2010** [comment]
Is this working or is there a velocity plugin?

**Comment by Skidamek** [comment]
Not yet, that's why it's open

**Comment by Barian0517** [comment]
is this done?

**Comment by Barian0517** [comment]
Is there any known way to use automodpack in a group server?

**Comment by Skidamek** [comment]
What does it mean?

**Comment by Barian0517** [comment]
Do you know if there’s a mod that allows Minecraft to load mods from folders outside the default mods directory at startup—kind of like how AutoModpack separates server mods from client mods?
I’d like to keep things organized and avoid mixing my personal mods with others.

**Comment by Skidamek** [comment]
Soon #409

**Comment by Skidamek** [comment]
velocity is now supported! https://modrinth.com/plugin/automodpackvelocity
